### DISCUSSION TITLE
Extending the components documentation for custom components

### DISCUSSION THREAD

**Discussion by bilalq** [cloudscape-design/components#968] [Q&A] [answered]
Hey everyone.

I first want to give a shoutout for getting this open-sourced. This is really an understated gem among Amazon's public contributions to open source. I've worked there in the past on several AWS consoles and experienced that evolution of the internal design systems first-hand. Seeing how far the team has come has been really impressive.

The feature I love the most is the documentation and playground for components. It's so much more cohesive of an experience than any Storybook setup or alternative I've ever used. However, there isn't really any documentation for extending the design system with custom components bespoke to a project/org/service/company/whatever. Ideally, I'd like to be able to generate my own build of the docs page with my own components on display similar to how [CloudScape's native ones are shown](https://cloudscape.design/components/button/?tabId=playground). Can you offer any guidance on doing that?

**Comment by rubencarvalho** [comment, marked as answer]
Hello Bilal,

We have a custom implementation of the playground using the [react-view](https://github.com/uber/react-view#readme) library.

Furthermore, we use the [typedoc](https://typedoc.org/) to generate the API documentation for our React components. 

Thank you,
Ruben

**Reply by bilalq** [reply]
Thanks for responding, @rubencarvalho. Would you consider opensourcing your playground setup and doc page build setup? I'd love to be able to host a fork of cloudscapes docs in full with extended custom components displayed.

**Reply by bilalq** [reply]
Pinging on this ask, @rubencarvalho.
